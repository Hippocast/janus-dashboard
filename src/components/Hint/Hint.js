import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';

import './Hint.css';

const b = block('j-hint');

const propTypes = {
    children: PropTypes.string.isRequired,
    title: PropTypes.bool,
};

const Hint = ({ children, title }) => (
    <div className={b({ title })()}>{ children }</div>
);

Hint.propTypes = propTypes;

export default Hint;
