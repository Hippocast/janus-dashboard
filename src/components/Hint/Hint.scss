@import 'index.scss';

.j-hint {
    font-size: $hf-font-size-small;
    color: $hf-color-gray-medium;
    line-height: $hf-line-height-medium;

    &--title {
        margin-bottom: $hf-gutter;
    }
}
