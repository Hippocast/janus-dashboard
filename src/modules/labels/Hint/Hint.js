import React from 'react';
import PropTypes from 'prop-types';

import './Hint.css';

const propTypes = {
    children: PropTypes.string.isRequired,
};

const Hint = ({ children }) => (
    <div className="j-hint">{ children }</div>
);

Hint.propTypes = propTypes;

export default Hint;
