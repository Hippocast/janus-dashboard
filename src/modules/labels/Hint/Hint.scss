@import 'index.scss';

.j-hint {
    font-size: $hf-font-size-small;
    color: $hf-color-gray-medium;
    line-height: $hf-line-medium;
}
