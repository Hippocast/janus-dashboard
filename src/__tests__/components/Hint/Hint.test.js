/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Hint from '../../../components/Hint/Hint'

describe('Hint component', () => {
  it('renders correctly', () => {
    const passedProps = {
      children: 'mock-text'
    }

    const tree = renderer
      .create(
        <Hint
          {...passedProps}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
