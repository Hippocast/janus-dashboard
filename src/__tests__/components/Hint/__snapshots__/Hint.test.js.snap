// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint component renders correctly 1`] = `
<div
  className="j-hint"
>
  mock-text
</div>
`;
